### PR TITLE
Move re_renderer examples to its own crate in order to make workspace level examples less confusing

### PR DESCRIPTION
**Pull request by Wumpf**
### What

This way we don't show renderer examples like "the examples". Also, due to some dependency issues with the optional re_renderer arrow dependency (details somewhat unclear..), we were previously no longer able to run examples on the web. This works nicely now again:

image

depicted:
* `cargo run -p re_renderer_examples --bin multiview` (native)
* `cargo run-wasm -p re_renderer_examples --bin multiview --features "webgl"` (webgl, using firefox here)
* `cargo run-wasm -p re_renderer_examples --bin multiview` (webgpu, using chrome here)

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Full build: [app.rerun.io](https://app.rerun.io/pr/4472/index.html)
  * Partial build: [app.rerun.io](https://app.rerun.io/pr/4472/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json) - Useful for quick testing when changes do not affect examples in any way
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4472)
- [Docs preview](https://rerun.io/preview/504ce9714e120af91374751e7ae86e70b195044c/docs) 
- [Examples preview](https://rerun.io/preview/504ce9714e120af91374751e7ae86e70b195044c/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -117,6 +117,7 @@ Update instructions:
 | rerun_c   | Rerun C SDK                          |
 | rerun_py  | Rerun Python SDK                     |
 | re_sdk    | Rerun logging SDK                    |
+<!-- | re_renderer_examples    | Examples for re_renderer | -->
```
Comment (teh-cmc) [inline ARCHITECTURE.md:120]: Hmm, what's this?

**Review by teh-cmc — APPROVED** [review]
Oh man I completely forgot about these; the memories :smiling_face_with_tear: 

Anyway, looks nice! Here's a crazy idea though: can we move `re_renderer_examples` under `re_renderer`?
As in, it would still be an independent crate, with a cargo name of `re_renderer_examples` but a repo/filesystem name of `re_renderer/examples`?

File: ARCHITECTURE.md (hunk 1 above)
Comment (Wumpf) [reply ARCHITECTURE.md:120]: our ci wants now all crates to be listed here, but I figured that's a weird one to list. So I pretend list it!

**Comment by Wumpf** [comment]
> As in, it would still be an independent crate, with a cargo name of re_renderer_examples but a repo/filesystem name of re_renderer/examples?

👍 I'll try that! Not sure if cargo gets confused by this

**Comment by Wumpf** [comment]
tried and Cargo doesn't like it at all. We figured it's time to directorify crates, not happening in this PR though.

**Pull request merged**